### ISSUE TITLE
[0.9.6.1]after restore,there is no data

### ISSUE THREAD

**Issue by lynch0227** [influxdata/influxdb#5137]
Hi guys, 
  when I restore influxdb, I can't query any data. Do I miss something?

  Here is the steps:
  1. service influxdb stop
  2. influxd restore -config /usr/local/influxdb/conf/influxdb.conf /tmp/mysnapshot
  3. chown -R influxdb: /usr/local/influxdb/data /usr/local/influxdb/meta
  4. service influxdb start

here is the restore log:
unpacking: _internal/monitor/1 (27365376 bytes)
unpacking: _internal/monitor/20 (5099520 bytes)
unpacking: _internal/monitor/21 (5468160 bytes)
unpacking: business_monitor/business_monitor_policy/10 (163840 bytes)
unpacking: business_monitor/business_monitor_policy/11 (155648 bytes)
unpacking: business_monitor/business_monitor_policy/12 (135168 bytes)
unpacking: business_monitor/business_monitor_policy/13 (155648 bytes)
unpacking: business_monitor/business_monitor_policy/14 (155648 bytes)
unpacking: business_monitor/business_monitor_policy/15 (155648 bytes)
unpacking: business_monitor/business_monitor_policy/16 (73728 bytes)
unpacking: business_monitor/business_monitor_policy/17 (114688 bytes)
unpacking: business_monitor/business_monitor_policy/18 (40960 bytes)
unpacking: business_monitor/business_monitor_policy/19 (28672 bytes)
unpacking: business_monitor/business_monitor_policy/2 (143360 bytes)
unpacking: business_monitor/business_monitor_policy/3 (159744 bytes)
unpacking: business_monitor/business_monitor_policy/4 (155648 bytes)
unpacking: business_monitor/business_monitor_policy/5 (135168 bytes)
unpacking: business_monitor/business_monitor_policy/6 (135168 bytes)
unpacking: business_monitor/business_monitor_policy/7 (135168 bytes)
unpacking: business_monitor/business_monitor_policy/8 (45056 bytes)
unpacking: business_monitor/business_monitor_policy/9 (135168 bytes)
unpacking: meta (1001 bytes)
[metastore] 2015/12/16 11:36:33 Using data dir: /usr/local/influxdb/meta
[metastore] 2015/12/16 11:36:34 Node at 127.0.0.1:8088 [Follower]
[metastore] 2015/12/16 11:36:36 Node at 127.0.0.1:8088 [Leader]. peers=[]
[metastore] 2015/12/16 11:36:36 Created local node: id=1, host=127.0.0.1:8088
[metastore] 2015/12/16 11:36:36 spun up monitoring for 1
[metastore] 2015/12/16 11:36:36 exec listener accept error and closed: listener closing
[metastore] 2015/12/16 11:36:36 RPC listener accept error and closed: listener closing
restore complete using /tmp/mysnapshot[influxdb@localhost ~]$

**Comment by bovy89** [comment]
Hi, I have made a couple of tests about snapshot..I hope this could help.
(I have omitted the restore log because it's equal to your, for all tests)

```
#influx
>create database test with duration 4w
>use test
>insert disk_free,hostname=server01 value=1 1451048988
>insert disk_free,hostname=server01 value=2 1451048989
>insert disk_free,hostname=server01 value=3 1451048990

#ll /var/lib/influxdb/wal/test/default/11/*
-rw-r--r--. 1 influxdb influxdb 127 Dec 25 15:50 /var/lib/influxdb/wal/test/default/11/000000.meta
```
The .meta file is not empty and contains my data

```
#influxd backup /root/snap1
```

Wait for "Metadata flush took 5.032419ms"

```
#ll /var/lib/influxdb/wal/test/default/11/*
-rw-r--r--. 1 influxdb influxdb 0 Dec 25 16:00 /var/lib/influxdb/wal/test/default/11/000001.meta

#influxd backup /root/snap2
```

Now I tried to restore snaphots:

- SNAP1:
```
service influxdb stop
influxd restore -config /etc/influxdb/influxdb.conf /root/snap1
chown -R influxdb:influxdb /var/lib/influxdb/*
service influxdb start
```

<b>---> db is present, but with no data</b>

- SNAP2:
```
service influxdb stop
influxd restore -config /etc/influxdb/influxdb.conf /root/snap2
chown -R influxdb:influxdb /var/lib/influxdb/*
service influxdb start
```
<b>---> db is present with my data</b>

**Comment by beckettsean** [comment]
backup and restore have been rewritten in 0.10+, no code fixes for 0.9 are planned.
